### PR TITLE
(fix) Controller: clear mapping pointer when unloading a mapping

### PR DESCRIPTION
**Pull request by ronso0**
With #13842 I noticed that the controller's pMapping still points to the previously loaded mapping when loading 'No mapping'.
Is there any reason for that?

### CODE REVIEW

File: src/controllers/bulk/bulkcontroller.h [hunk 1]
```diff
@@ -43,6 +43,9 @@ class BulkController : public Controller {
 
     virtual std::shared_ptr<LegacyControllerMapping> cloneMapping() override;
     void setMapping(std::shared_ptr<LegacyControllerMapping> pMapping) override;
+    void clearMapping() override {
+        m_pMapping = nullptr;
+    }
```
Comment (Swiftb0y) [inline src/controllers/bulk/bulkcontroller.h:48]: just call `setMapping` with a `nullptr` directly.

Comment (ronso0) [reply src/controllers/bulk/bulkcontroller.h:48]: That was the first thing I tried.
Gives me this when selecting No mapping
```
DEBUG ASSERT: "pMapping.use_count() == 1" in function std::shared_ptr<_Tp> Controller::downcastAndTakeOwnership(std::shared_ptr<LegacyControllerMapping>&&) [with SpecificMappingType = LegacyMidiControllerMapping] at ./src/controllers/controller.h:89
```

Comment (ronso0) [reply src/controllers/bulk/bulkcontroller.h:48]: https://github.com/mixxxdj/mixxx/blob/815219d8a048165afd60c5283874da71f6f3e342/src/controllers/controller.h#L82-L96

Comment (Swiftb0y) [reply src/controllers/bulk/bulkcontroller.h:48]: right, that function does not take that usecase into account. Simply add a `if (pMapping == nullptr) { return nullptr; }` so it supports that usecase. Alternatively you can add the appropriate handling to the `setMapping` implementations.

Comment (ronso0) [reply src/controllers/bulk/bulkcontroller.h:48]: Okay. I only took a brief look at this and didn't want to touch it.
But yeah, that's a straight forward solution.
Thanks!

**Review by Swiftb0y — APPROVED** [review]
conceptually LGTM. Thank you. Waiting for CI.

**Comment by ronso0** [comment]
Ready for merge, right?

**Review by daschuer — APPROVED** [review]
Thank you. Maybe this also fixes https://github.com/mixxxdj/mixxx/issues/13940? 
https://github.com/mixxxdj/mixxx/pull/13980

**Comment by ronso0** [comment]
Hmm I dont see how this could be related.

**Comment by daschuer** [comment]
Yes, probably, not but at least it touches the same code.

**Pull request merged**